purchase_requisition_line_description/models/purchase_requisition_line.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class PurchaseRequisitionLine(models.Model):
+    _inherit = "purchase.requisition.line"
+
+    name = fields.Text()
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        desc = ""
+        res = super(PurchaseRequisitionLine, self)._onchange_product_id()
+        if self.product_id:
```

Comment:
```suggestion
        desc = self.name
        if self.product_id:
```

Reply:
@i-vyshnevska  that won't work. I already checked it yesterday with @grindtildeath , somehow self.name is type bool. If later we try to concat str with bool we run into issues with TypeError Invalid operand. Coulnd't find out why it is bool type instead of 'str', so we made sure that desc is string by assigning empty string to it and update self.name with desc in latest step.


```
File "/odoo/external-src/purchase-workflow/purchase_requisition_line_description/models/purchase_requisition_line.py", line 28, in _onchange_product_id
    desc += '[' + vendor_data.product_code + '] '
TypeError: unsupported operand type(s) for +=: 'bool' and 'str'
```

Reply:
desc = (self.name or "")

Reply:
works fine, thank you

Reply:
@i-vyshnevska  with this once you have line with product and you update it you get description not properly updated (contains still data from product before change):
![image](https://user-images.githubusercontent.com/59824990/114893802-594ecf80-9e0e-11eb-890c-c472e12ed6b9.png)
so currently it works fine only if your name is empty. i'll revert it back.